e2e_tests/tests/cluster/test_users.py

```diff
@@ -266,61 +265,107 @@ def test_logout(clean_auth: None) -> None:
-    creds = create_test_user(ADMIN_CREDENTIALS, True)
+def test_activate_deactivate(clean_auth: None, login_admin: None) -> None:
+    creds = create_test_user(True)
 
     # Make sure we can log in as the user.
     assert log_in_user(creds) == 0
 
     # Log out.
     assert log_out_user() == 0
 
+    # login admin again.
+    log_in_user(ADMIN_CREDENTIALS)
+
     # Deactivate user.
-    assert activate_deactivate_user(False, creds.username, ("admin", "")) == 0
+    assert activate_deactivate_user(False, creds.username) == 0
 
     # Attempt to log in again.
     assert log_in_user(creds) != 0
 
     # Activate user.
-    assert activate_deactivate_user(True, creds.username, ("admin", "")) == 0
+    assert activate_deactivate_user(True, creds.username) == 0
+
+    # Now log in again.
+    assert log_in_user(creds) == 0
+
+    # SDK testing for activating and deactivating.
+    log_in_user(ADMIN_CREDENTIALS)
+    det_obj = Determined(master=conf.make_master_url())
+    user = det_obj.get_user_by_name(user_name=creds.username)
+    assert not user.deactivate().user.active
+    assert user.activate().user.active
 
     # Now log in again.
     assert log_in_user(creds) == 0
 
 
 @pytest.mark.e2e_cpu
 @pytest.mark.e2e_cpu_postgres
-def test_change_password(clean_auth: None) -> None:
+def test_change_password(clean_auth: None, login_admin: None) -> None:
     # Create a user without a password.
-    creds = create_test_user(ADMIN_CREDENTIALS, False)
+    creds = create_test_user(False)
 
     # Attempt to log in.
     assert log_in_user(creds) == 0
 
     # Log out.
     assert log_out_user() == 0
 
-    newPassword = get_random_string()
-    assert change_user_password(creds.username, newPassword, ADMIN_CREDENTIALS) == 0
+    # login admin
+    log_in_user(ADMIN_CREDENTIALS)
 
+    newPassword = get_random_string()
+    assert change_user_password(creds.username, newPassword) == 0
     assert log_in_user(authentication.Credentials(creds.username, newPassword)) == 0
 
+    newPasswordSdk = get_random_string()
+    det_obj = Determined(master=conf.make_master_url())
+    user = det_obj.get_user_by_name(user_name=creds.username)
+    assert user.change_password(new_password=newPasswordSdk)
+    assert log_in_user(authentication.Credentials(creds.username, newPasswordSdk)) == 0
+
+
+@pytest.mark.e2e_cpu
+def test_change_username(clean_auth: None, login_admin: None):
+    creds = create_test_user()
+    new_username = "renameuser4"
+    print("new username")
+    print(creds.username)
+    print(new_username)
+    command = ["det", "-m", conf.make_master_url(), "user", "rename", creds.username, new_username]
+    assert subprocess.call(command) == 0
+    det_obj = Determined(master=conf.make_master_url())
+    user = det_obj.get_user_by_name(user_name=new_username)
+    assert user.username == new_username
+    assert log_in_user(authentication.Credentials(new_username, "")) == 0
+
+    # Test SDK
+    new_username = "renameuser5"
```

Comment:
can we sometimes throw in users with non-alphanumeric names? e.g. `rename-user-$5`
I recently discovered there was a bug prohibiting that *in rename only* so testing that would be nice.

Reply:
Messaged about this on slack.